src/Resolvers/Microsoft.DotNet.NativeWrapper/EnvironmentProvider.cs

```diff
@@ -51,9 +53,7 @@ public string GetCommandPath(string commandName)
         {
             var commandNameWithExtension = commandName + Constants.ExeSuffix;
             var commandPath = SearchPaths
-                .Where(p => !Path.GetInvalidPathChars().Any(c => p.Contains(c)))
-                .Select(p => Path.Combine(p, commandNameWithExtension))
-                .FirstOrDefault(File.Exists);
+                .FirstOrDefault(p => File.Exists(Path.Combine(p, commandNameWithExtension)));
```

Comment:
This now returns the search path without the commandNameWithExtension, whereas it previously included the commandNameWithExtension, right? Should we add one more Path.Combine wrapping this commandPath before returning?

Reply:
Great catch! Fixed, thank you.